### PR TITLE
Feature obsdef 15211 side navigation

### PR DESCRIPTION
**Pull request by ShitalSD**
Side navigation wrapper
Test link 
http://10.249.253.4:6008/

Screenshot (495)

Screenshot (491)
Screenshot (494)

### CODE REVIEW

File: src/styles/common/_mixins.scss [hunk 1]
```diff
@@ -178,3 +178,24 @@
...
+    @if($isSubSection) {
+        color: $side-nav-trigger-active-color!important;
+        font-weight: $font-weight-medium!important;
+        border-left: 0.25rem solid $side-nav-select-bar-color;
+        .nav-icon {
+            fill: inherit;
+        }
+    } @else if($isSelectBarRequired) {
+        @extend %side-nav-active-no-subsection;
+        border-left: 0.25rem solid $side-nav-select-bar-color;
```
Comment (akate1) [inline src/styles/common/_mixins.scss:196]: is it possible to create constant for ` 0.25rem solid $side-nav-select-bar-color;` ?

File: src/styles/components/VerticalNav.scss [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                padding-left: 2.2rem;
+                .nav-text {
+                   @extend %side-nav-child-item-height;
+                }
+                &:active {
+                    @include side-nav-active(false, false);
+                }
+                &.active {
+                    @include side-nav-active(false, false);
+                }
```
Comment (akate1) [inline src/styles/components/VerticalNav.scss:61]: is it possible to merge these into a single class/entity?

File: src/styles/components/VerticalNav.scss [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            }   
+       }
+    }
+    .nav-group-content {
+        .nav-group-trigger {
+           @extend %side-nav-item-height;
+        }
+        .nav-group-text {
+            @extend %side-nav-item-height;
+        }
```
Comment (akate1) [inline src/styles/components/VerticalNav.scss:71]:
```suggestion
        .nav-group-trigger , .nav-group-text{
           @extend %side-nav-item-height;
        }
 ```

File: src/styles/components/VerticalNav.scss [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            .nav-group-trigger-icon {
+                color: $side-nav-trigger-icon-color;
+            }
+        }
+        &:active {
+           @include side-nav-active(true, true);
+        }
+        &.active {
+            @include side-nav-active(true, true);
+        }
```
Comment (akate1) [inline src/styles/components/VerticalNav.scss:84]:
```suggestion
        &:active,  &.active {
           @include side-nav-active(true, true);
        }
        
```

**Review by akate1 — COMMENTED** [review]
@ShitalSD can you please update version in package.json ? so that we can publish new version to npm.

File: src/styles/common/_mixins.scss (hunk 1 above)
Comment (ShitalSD) [reply src/styles/common/_mixins.scss:196]: Done

File: src/styles/components/VerticalNav.scss (hunk 2 above)
Comment (ShitalSD) [reply src/styles/components/VerticalNav.scss]: Done

File: src/styles/components/VerticalNav.scss (hunk 3 above)
Comment (ShitalSD) [reply src/styles/components/VerticalNav.scss]: Done

File: src/styles/components/VerticalNav.scss (hunk 4 above)
Comment (ShitalSD) [reply src/styles/components/VerticalNav.scss]: Done

**Review by akate1 — APPROVED** [review]

File: package.json [hunk 5]
```diff
@@ -162,4 +162,4 @@
     "snapshotResolver": "<rootDir>/testResolver.tsx",
     "testEnvironment": "jest-environment-jsdom-fourteen"
   }
-}
+}
```
Comment (m-manaswini) [inline package.json:165]: Please add new line

**Review by m-manaswini — APPROVED** [review]
LGTM

**Pull request merged**